pkg/log/zap/flags.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package zap contains helpers for setting up a new logr.Logger instance
+// using the Zap logging framework.
+
+package zap
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type encoderConfigFunc func(*zapcore.EncoderConfig)
+
+type encoderValue struct {
+	set     bool
+	setFunc func(zapcore.Encoder)
+	str     string
+}
+
+func (v encoderValue) String() string {
+	return v.str
+}
+
+func (v encoderValue) Type() string {
+	return "encoder"
+}
+
+func (v *encoderValue) Set(e string) error {
+	v.set = true
+	val := strings.ToLower(e)
+	switch val {
+	case "json":
+		v.setFunc(newJSONEncoder())
+	case "console":
+		v.setFunc(newConsoleEncoder())
+	default:
+		return fmt.Errorf("invalid encoder value \"%s\"", e)
+	}
+	v.str = e
+	return nil
+}
+
+func newJSONEncoder(ecfs ...encoderConfigFunc) zapcore.Encoder {
```

Comment:
In operator-sdk, we're using `encoderConfigFunc` to allow other flags to control other aspects of the log encoding (e.g. [the timestamp format](https://github.com/operator-framework/operator-sdk/blob/633b92c888ad3eeb9704441b7f6d6bd15f933d0e/pkg/log/zap/flags.go#L211-L215)). We're not doing that here (yet?), so maybe we should remove the code that uses it for now?